core/src/SYCL/Kokkos_SYCL_TeamPolicy.hpp

```diff
@@ -170,10 +155,10 @@ class Kokkos::Impl::TeamPolicyInternal<Kokkos::SYCL, Properties...>
       : m_space(space_),
         m_league_size(league_size_),
         m_team_size(team_size_request),
-        m_vector_length(
-            (vector_length_request > 0)
-                ? verify_requested_vector_length(vector_length_request)
-                : (verify_requested_vector_length(1))),
+        m_vector_length((vector_length_request > 0)
+                            ? Kokkos::bit_floor(std::min<unsigned>(
+                                  vector_length_request, vector_length_max()))
+                            : 1),
```

Comment:
@masterleinad thoughts about doing
```suggestion
        m_vector_length(Kokkos::bit_floor(std::clamp<unsigned>(
                                  vector_length_request, 1, vector_length_max()))),
```
instead?

Reply:
I have slight preference for the latter.

Reply:
Done